options.go

```diff
@@ -57,6 +58,19 @@ func WithColumnMap(m map[string]string) Option {
 	}
 }
 
+// Add the new option function
+// WithColumnFieldTag provides an option to use a specific struct tag for field mapping
+// If a field has this tag, the tag value will be used instead of the field name
+func WithColumnFieldTag(tagName string) Option {
+	return func(o *options) error {
+		if tagName == "" {
+			return fmt.Errorf("mql.WithColumnFieldTag: empty tag name: %w", ErrInvalidParameter)
+		}
```

Comment:
Should we validate that there's a field name in the tag here vs waiting till we
try to use the tag in `validate(...)`?

Reply:
That would mean passing model to test that? This method currently applies to all models. Unless I am misunderstanding the ask

Reply:
Yes, but now that I'm thinking about it, that's not reasonable.